sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -148,9 +148,17 @@ class SessionCatalog(
     currentDb = db
   }
 
-  def getDefaultDBPath(db: String): String = {
+  /** Get the path for creating a non-default database. */
+  def createDatabasePath(db: String, path: Option[String]): String = {
     val database = if (conf.caseSensitiveAnalysis) db else db.toLowerCase
-    new Path(new Path(conf.warehousePath), database + ".db").toString
+    val dbPath = path.map(new Path(_)).getOrElse {
+      val defaultPath = conf.warehousePath
+      if (org.apache.commons.lang.StringUtils.isBlank(defaultPath)) {
+        throw new AnalysisException("spark.sql.warehouse.dir is blank")
```

Comment:
This is from what Hive metastore is doing. 
https://github.com/apache/hive/blob/134b6cccbd7237901f7f7594626796863ca0150a/metastore/src/java/org/apache/hadoop/hive/metastore/Warehouse.java#L74-L77

I also have the same question like you. Let me remove it since we do not follow what Hive is doing.